lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1330,11 +1331,11 @@ export class UserAgentApplication {
             window.renewStates.push(serverAuthenticationRequest.state);
         }
 
-        // note: scope here is clientId
-        this.registerCallback(serverAuthenticationRequest.state, this.clientId, resolve, reject);
+        const requestUid = RequestUtils.computeTokenRequestUniquenessIdentifier(serverAuthenticationRequest, account);
+        this.registerCallback(serverAuthenticationRequest.state, requestUid, resolve, reject);
         this.logger.infoPii("Navigate to:" + urlNavigate);
         frameHandle.src = "about:blank";
-        this.loadIframeTimeout(urlNavigate, frameName, this.clientId).catch(error => reject(error));
+        this.loadIframeTimeout(urlNavigate, frameName, requestUid).catch(error => reject(error));
```

Comment:
this previous logic around using the clientId as a scope seems a bit hacky. I noticed that the ctor for ServerRequestParameters will initialize the scopes array to the clientId if the request does not include it. Curious, why wasn't a scope of "openid" used for this flow? this is probably something to revist. In any case, this PR should continue to accommodate that logic.